src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -808,21 +850,28 @@ private static PlatformAttributes CopyAllAttributes(PlatformAttributes copyTo, P
 
                 AddPlatformAttributes(symbol.GetAttributes(), ref attributes);
 
+                if (symbol is IMethodSymbol method && (method.MethodKind == MethodKind.PropertyGet || method.MethodKind == MethodKind.PropertySet))
+                {
+                    // Add attributes for the associated Property
+                    AddPlatformAttributes(method.AssociatedSymbol.GetAttributes(), ref attributes);
+                }
+
                 attributes = platformSpecificMembers.GetOrAdd(symbol, attributes);
             }
 
             return attributes != null;
 
             static bool AddPlatformAttributes(ImmutableArray<AttributeData> immediateAttributes, [NotNullWhen(true)] ref SmallDictionary<string, PlatformAttributes>? attributes)
             {
+                bool added = false;
                 foreach (AttributeData attribute in immediateAttributes)
                 {
                     if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
                     {
-                        TryAddValidAttribute(ref attributes, attribute);
+                        added |= TryAddValidAttribute(ref attributes, attribute);
```

Comment:
Does this skip calls to TryAddValidAttribute if added is already true? I always get confused with semantics of such compound or operations. Might be simpler to just write this as an if whose body sets added to true unconditionally.

Reply:
Btw, I did not see implementation for TryAddValidAttribute change in this PR. Do we need the changes to AddPlatformAttribues?

Reply:
The value returned from this function is not really used anywhere, I added this change to resolve the symbols returned in Enumerable, just `attributes != null` is not really express if any attribute added. But later choose another approach for Enumerable. But returning correct value will be helpful later when we refactor the logic to use immutable attributes,